### PR TITLE
add PrestoOperator

### PR DESCRIPTION
**Pull request by eladkal**
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by mik-laj** [comment]
Are these operators really working? I had a problem with the run method.
Please look at: https://github.com/apache/airflow/blob/master/tests/providers/google/cloud/operators/test_presto_to_gcs_system.py#L137-L141

Could you add system tests and a sample DAG? This would make it easier to test these operators.
You can base on:
https://github.com/apache/airflow/blob/master/tests/providers/google/cloud/operators/test_presto_to_gcs_system.py

File: airflow/providers/presto/operators/presto.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        super().__init__(*args, **kwargs)
+        self.presto_conn_id = presto_conn_id
+        self.sql = sql
+        self.parameters = parameters
+        self.catalog = catalog
+        self.schema = schema
+
+    def execute(self, context):
+        self.log.info('Executing: %s', self.sql)
+        hook = PrestoHook(presto_conn_id=self.presto_conn_id,
```
Comment (mik-laj) [inline airflow/providers/presto/operators/presto.py:66]: Can you look at this PR?
https://github.com/apache/airflow/pull/3891/files

**Comment by eladkal** [comment]
@mik-laj yes it's working. I do have some work to fix the tests and add examples but the operator runs on my presto cluster. It does has limitations where it can't execute batch statements.

as for your concern:
`                # Presto does not execute queries until the result is fetched. :-(`
The reason is explained at https://github.com/prestodb/presto/pull/10603

File: airflow/providers/presto/operators/presto.py (hunk 1 above)
Comment (eladkal) [reply airflow/providers/presto/operators/presto.py:66]: It's not quite the same. PrestoHook behave differently than other DBs hooks. For example the `autocommit`. I agree that SQLOperator is preferred but it's something I need time to look into. Might worth revising this in the future.

**Pull request closed without merging**